pkg/tasks/tasks.go

```diff
@@ -21,35 +21,64 @@ import (
 	"k8s.io/klog/v2"
 )
 
+// TaskRunner manages lists of task groups. Through the RunAll method task groups are
+// executed, the groups sequentially, each group of tasks concurrently.
 type TaskRunner struct {
-	client *client.Client
-	tasks  []*TaskSpec
+	client     *client.Client
+	taskGroups []*TaskGroup
 }
 
-func NewTaskRunner(client *client.Client, tasks []*TaskSpec) *TaskRunner {
+// NewTaskRunner returns a task runner. tasks is the first task group that will
+// be executed, before any list added via AppendTaskGroup.
+func NewTaskRunner(client *client.Client, tasks *TaskGroup) *TaskRunner {
```

Comment:
We could make it a variadic function and get rid of `AppendTaskGroup`?

```suggestion
func NewTaskRunner(client *client.Client, taskGroups ...*TaskGroup) *TaskRunner {
```

Reply:
:+1: Indeed we could...and probably should too.